Test flakiness: daemon webui port-bind race and community-create near-timeout on Windows

Two unrelated flakes surfaced during the CI runs on PR #41. They are independent of that PR's code change — the same commit was green on the rerun without any code changes. Captured here so we can root-cause them rather than just live with the noise.

## Flake 1 — Ubuntu: 5chan webui ECONNREFUSED

**Test:** \`test/cli/daemon.test.ts:774\` — \`bitsocial daemon webui > 5chan webui does not contain the hash redirect script\`

**Symptom:**
\`\`\`
TypeError: fetch failed
Caused by: Caused by: AggregateError
  - connect ECONNREFUSED 127.0.0.1:39138
  - connect ECONNREFUSED ::1:39138
\`\`\`

**Failing run:** https://github.com/bitsocialnet/bitsocial-cli/actions/runs/26142249527/job/76889997036 (since-passed on rerun)

**Likely cause:** The \`beforeAll\` calls \`startPkcDaemon(...)\` and resolves, but the HTTP listener on \`localhost:39138\` (the PKC RPC port) isn't actually accepting connections yet when the \`fetch\` runs. So either:
- \`startPkcDaemon\` returns before the HTTP server has finished binding (race between \"daemon ready\" signal and listener \`.listen\` completion), **or**
- The HTTP listener crashed during startup on this run.

**Investigation suggestion:** In \`test/helpers/daemon-helpers.ts\` (or wherever \`startPkcDaemon\` lives), make readiness wait on a \`fetch(rpcUrl + '/')\` probe returning OK, not just on a log line / pubsub-ready signal. The other daemon tests that *do* probe HTTP after startup pass consistently — this one only fetches once and trusts \`beforeAll\`.

## Flake 2 — Windows: community create just barely times out

**Test:** \`test/cli/command-completion-time.test.ts:125\` — \`CLI commands complete within 10s (real pkc instance) > community create completes within 10s\`

**Symptom:** Timed out at 10018ms (baseline on healthy Windows runs: ~4455ms; on Ubuntu: ~1979ms).

**Failing run:** https://github.com/bitsocialnet/bitsocial-cli/actions/runs/26142249527/job/76889997061 (since-passed on rerun in 13m35s)

**Likely cause:** The 10s budget has too little headroom for Windows runners under load. We've previously seen Windows runs roughly 2-3x slower than Linux. A test budget that fits in 2s on Linux + 4.5s on a fast Windows runner has no slack for a slow runner.

**Investigation suggestion:** Either:
- Bump the timeout for Windows specifically (\`process.platform === 'win32' ? 20_000 : 10_000\`), **or**
- Measure where the 5-10s is actually spent and shave whatever's avoidable. If most of it is daemon-handshake / pubsub warmup that the test doesn't care about, the test could pre-warm before \`it()\` starts.

## Why this isn't pkc-js

PR #41 had two commits — first the logs fix alone (green on all 3 platforms), then the logs fix + pkc-js 0.0.34 → 0.0.35 bump (these two flakes). The same SHA passed on rerun with no code changes, ruling out the bump as the cause.

--- a/src/webui/daemon-server.ts
+++ b/src/webui/daemon-server.ts
@@ -46,7 +46,23 @@ export async function startDaemonServer(rpcUrl: URL, ipfsGatewayUrl: URL, pkcOpt
     // Start pkc-js RPC
     const log = PKCLogger("bitsocial-cli:daemon:startDaemonServer");
     const webuiExpressApp = express();
-    const httpServer = webuiExpressApp.listen(Number(rpcUrl.port));
+    // Wait for bind to actually complete before returning. Calling express.listen() without
+    // awaiting 'listening' lets startup proceed before the port is accepting connections,
+    // and without an 'error' handler a bind failure becomes an uncaughtException that kills
+    // the daemon *after* it has already logged "Communities in data path" — see issue #42.
+    const httpServer = await new Promise<import("http").Server>((resolve, reject) => {
+        const server = webuiExpressApp.listen(Number(rpcUrl.port));
+        const onListening = () => {
+            server.off("error", onError);
+            resolve(server);
+        };
+        const onError = (err: Error) => {
+            server.off("listening", onListening);
+            reject(err);
+        };
+        server.once("listening", onListening);
+        server.once("error", onError);
+    });
     log("HTTP server is running on", "0.0.0.0" + ":" + rpcUrl.port);
     const rpcAuthKey = await _generateRpcAuthKeyIfNotExisting(pkcOptions.dataPath!);
     const PKCRpc = await import("@pkcprotocol/pkc-js/rpc");

--- a/test/webui/daemon-server.test.ts
+++ b/test/webui/daemon-server.test.ts
@@ -11,11 +11,11 @@ import { startDaemonServer } from "../../dist/webui/daemon-server.js";
 // The contract we want: if the port can't be bound, startDaemonServer must reject.
 describe("startDaemonServer bind contract", () => {
     it("rejects when the RPC port is already taken (regression for issue #42)", async () => {
-        // Blocker must bind to 0.0.0.0 because express.listen(port) (no host arg) also
-        // binds to 0.0.0.0. On BSD-derived stacks (macOS, Windows) a wildcard bind and
-        // a 127.0.0.1 bind on the same port can coexist, so binding the blocker to
-        // 127.0.0.1 would let the daemon's wildcard bind succeed and the test wouldn't
-        // exercise the EADDRINUSE path.
+        // The blocker must bind the *same way* the daemon does — no host argument —
+        // so it lands on whatever default Node picks for this platform. On macOS and
+        // Windows that default is the IPv6 wildcard `::` with IPV6_V6ONLY=true, which
+        // does NOT conflict with a 0.0.0.0 bind. Binding the blocker to 127.0.0.1 or
+        // 0.0.0.0 would let the daemon's listen succeed there and never hit EADDRINUSE.
         const blocker = net.createServer();
         const port = await new Promise<number>((resolve, reject) => {
             blocker.once("listening", () => {
@@ -24,7 +24,7 @@ describe("startDaemonServer bind contract", () => {
                 else reject(new Error(`unexpected address ${JSON.stringify(addr)}`));
             });
             blocker.once("error", reject);
-            blocker.listen(0, "0.0.0.0");
+            blocker.listen(0);
         });
 
         // Swallow any stray uncaughtException emitted by an unguarded server.listen()

--- a/test/webui/daemon-server.test.ts
+++ b/test/webui/daemon-server.test.ts
@@ -11,6 +11,11 @@ import { startDaemonServer } from "../../dist/webui/daemon-server.js";
 // The contract we want: if the port can't be bound, startDaemonServer must reject.
 describe("startDaemonServer bind contract", () => {
     it("rejects when the RPC port is already taken (regression for issue #42)", async () => {
+        // Blocker must bind to 0.0.0.0 because express.listen(port) (no host arg) also
+        // binds to 0.0.0.0. On BSD-derived stacks (macOS, Windows) a wildcard bind and
+        // a 127.0.0.1 bind on the same port can coexist, so binding the blocker to
+        // 127.0.0.1 would let the daemon's wildcard bind succeed and the test wouldn't
+        // exercise the EADDRINUSE path.
         const blocker = net.createServer();
         const port = await new Promise<number>((resolve, reject) => {
             blocker.once("listening", () => {
@@ -19,7 +24,7 @@ describe("startDaemonServer bind contract", () => {
                 else reject(new Error(`unexpected address ${JSON.stringify(addr)}`));
             });
             blocker.once("error", reject);
-            blocker.listen(0, "127.0.0.1");
+            blocker.listen(0, "0.0.0.0");
         });
 
         // Swallow any stray uncaughtException emitted by an unguarded server.listen()

--- a/test/webui/daemon-server.test.ts
+++ b/test/webui/daemon-server.test.ts
@@ -1,0 +1,47 @@
+import { describe, it, expect } from "vitest";
+import net from "net";
+import { directory as randomDirectory } from "tempy";
+import { startDaemonServer } from "../../dist/webui/daemon-server.js";
+
+// Regression test for issue #42:
+// startDaemonServer used to call webuiExpressApp.listen(port) fire-and-forget — no
+// await on 'listening', no 'error' handler. If bind failed, the promise still resolved
+// and the bind error showed up later as an uncaughtException, crashing the daemon
+// AFTER the test helper had already accepted "Communities in data path" as readiness.
+// The contract we want: if the port can't be bound, startDaemonServer must reject.
+describe("startDaemonServer bind contract", () => {
+    it("rejects when the RPC port is already taken (regression for issue #42)", async () => {
+        const blocker = net.createServer();
+        const port = await new Promise<number>((resolve, reject) => {
+            blocker.once("listening", () => {
+                const addr = blocker.address();
+                if (addr && typeof addr === "object") resolve(addr.port);
+                else reject(new Error(`unexpected address ${JSON.stringify(addr)}`));
+            });
+            blocker.once("error", reject);
+            blocker.listen(0, "127.0.0.1");
+        });
+
+        // Swallow any stray uncaughtException emitted by an unguarded server.listen()
+        // so the test process survives long enough to assert the actual behavior.
+        const stray: Error[] = [];
+        const uncaughtHandler = (err: Error) => stray.push(err);
+        process.on("uncaughtException", uncaughtHandler);
+
+        try {
+            await expect(
+                startDaemonServer(
+                    new URL(`ws://127.0.0.1:${port}`),
+                    new URL("http://127.0.0.1:6754"),
+                    { dataPath: randomDirectory() }
+                )
+            ).rejects.toThrow(/EADDRINUSE|address already in use/i);
+            // And no stray uncaughtException either — the bind error must come back
+            // through the promise, not the process.
+            expect(stray).toEqual([]);
+        } finally {
+            process.off("uncaughtException", uncaughtHandler);
+            await new Promise<void>((resolve) => blocker.close(() => resolve()));
+        }
+    });
+});

